config.go

```diff
@@ -405,6 +422,31 @@ func loadConfig() (*config, []string, error) {
 		}
 		fmt.Fprintln(os.Stderr, err)
 		return nil, nil, err
+
+	} else if cfg.CreateTemp == true {
+		tempWalletExists := false
+
+		// Error if the wallet already exists.
+		if fileExists(dbPath) {
+			err := fmt.Errorf("The wallet already exists. Loading this " +
+				"wallet instead.")
+			fmt.Fprintln(os.Stderr, err)
+			tempWalletExists = true
+		}
+
+		if !tempWalletExists {
+			// Ensure the data directory for the network exists.
+			if err := checkCreateDir(netDir); err != nil {
```

Comment:
If you don't call err := checkCreateDir(netDir), if the datadir is missing the wallet will never initialize creation (createSimulateWallet throws an error).

The fileExists calling is just a method for if someone screws up and tries to create a wallet in a directory that already has one; it's loading it as a default behaviour. It could also fail by default, however cfg.Create above would then have inconsistent behaviour.